eras/shelley-ma/test-suite/src/Test/Cardano/Ledger/ShelleyMA/Serialisation/Roundtrip.hs

```diff
@@ -60,4 +97,118 @@ allEraRoundtripTests =
+        forAll genMultiAssetMintNonZero $
+          \ma -> roundTripCborExpectation ma
+    , testProperty "decode 0 as MultiAsset for pre-conway" $
+        forAll genMultiAssetMintZero $
+          \ma -> roundTripCborRangeExpectation minBound (natVersion @8) ma
+    , testProperty "fail decode 0 as MultiAsset for conway" $
+        forAll genMultiAssetMintZero $
+          \ma -> roundTripFailureExpectation (natVersion @9) ma
+    ]
+
+maryvalueCBORTests :: TestTree
+maryvalueCBORTests =
+  testGroup
+    "MaryValue CBOR roundtrips"
+    [ testProperty "decode (Positive Int) as MaryValue for all eras" $
+        forAll genMaryValuePositive $
+          \mv -> roundTripCborExpectation mv
+    , testProperty "fail decode (Negative Int) as MaryValue for all eras" $
+        forAll genMaryValueNegative $
+          \mv -> forM_ [minBound .. maxBound] $ \version -> roundTripFailureExpectation version mv
+    , testProperty "fail decode 0 as MaryValue for conway" $
+        forAll genMaryValueZero $
+          \mv -> roundTripFailureExpectation (natVersion @9) mv
+    , testProperty "decode 0 as MaryValue for pre-conway" $
+        forAll genMaryValueZero $
+          \mv -> roundTripCborRangeExpectation minBound (natVersion @8) mv
+    ]
+
+genCompactValue :: Gen Word64 -> Gen (CompactValue StandardCrypto)
+genCompactValue genWord64 = do
+  i <- genWord64
+  w32 <- arbitrary @Word32
+  sbs <- SBS.pack <$> resize 1 (listOf1 (arbitrary @Word8))
+  -- elements [CompactValueAdaOnly (CompactCoin i), CompactValueMultiAsset (CompactCoin i) (1 :: Word32) "a"]
+  elements [CompactValueAdaOnly (CompactCoin i), CompactValueMultiAsset (CompactCoin i) w32 sbs]
+
+compactvalueCBORTests :: TestTree
+compactvalueCBORTests =
+  testGroup
+    "CompactValue CBOR roundtrips"
+    [ testProperty "decode 0 as (CompactValue c = CompactCoin) for pre-conway" $
+        forAll (genCompactValue (pure 0)) $ \cv ->
+          roundTripCborRangeExpectation minBound (natVersion @8) cv
+    , testProperty "fail decode 0 as (CompactValue c = CompactCoin) for conway" $
+        forAll (genCompactValue (pure 0)) $
+          \cv -> roundTripFailureExpectation (natVersion @9) cv
+    , testProperty "decode (Positive Int) as (CompactValue c = CompactCoin) for all eras" $
+        forAll (genCompactValue (getPositive <$> arbitrary)) $
+          \cv -> roundTripCborExpectation cv
+    , -- (roundTripCborExpectation @(CompactValue StandardCrypto) (CompactValueAdaOnly (CompactCoin i)))
+      testProperty "fail decode (Negative Int) as (CompactValue c = CompactCoin) for all eras" $
+        forAll (genCompactValue (getNegative <$> arbitrary)) $ \cv ->
+          forM_ [minBound .. maxBound] $ \version ->
+            roundTripFailureExpectation version cv
+    ]
+
+valueCBORTests :: TestTree
+valueCBORTests =
+  testGroup
+    "Value CBOR roundtrips"
+    [ coinCBORTests
+    , multiassetCBORTests
+    , maryvalueCBORTests
+    -- , compactvalueCBORTests
```

Comment:
@lehins These tests are commented out because no matter how `CompactValue` is generated: via `toCompact (MultiAsset c)` or `genCompactValue`, they continue to result in `Heap exhausted`.

Reply:
Does each and every one of the four individually causes heap exhaustion?

Reply:
I have updated the code with commented out examples and `NOTE`s (inline) accompanying each case separately, in the hope that it can add more information for debugging.